### PR TITLE
std.socket: add scope to close

### PR DESCRIPTION
**Pull request by WalterBright**
In the service of making class destructors `scope`. See:

https://github.com/dlang/dmd/pull/14402

### CODE REVIEW

**Comment by WalterBright** [comment]
How on earth does std.socket cause std.path to fail?

**Comment by WalterBright** [comment]
The std.path failures persist even when I comment out the changes in this PR. The Phobos master is not working.

**Comment by RazvanN7** [comment]
```
std/socket.d(3644): Error: function `nothrow @nogc @trusted void std.socket.__unittest_L3588_C7.TestSocket.close()` does not override any function, did you mean to override `nothrow @nogc scope @trusted void std.socket.Socket.close()`?
posix.mak:386: recipe for target 'generated/linux/debug/32/unittest/std/socket.o' failed
```

File: std/socket.d [hunk 1]
```diff
@@ -2923,7 +2923,7 @@ public:
     }
 
 
-    private static void _close(socket_t sock) @system nothrow @nogc
+    private static void _close(scope socket_t sock) @system nothrow @nogc
```
Comment (dkorpel) [inline std/socket.d:2926]: `socket_t` is an `int`, no need to add `scope` here

Comment (WalterBright) [reply std/socket.d]: done

**Comment by dkorpel** [comment]
Tests fail because your local branch doesn't include https://github.com/dlang/phobos/pull/8547, please rebase so it's up to date

**Comment by WalterBright** [comment]
@dkorpel thanks for the tip!

**Comment by WalterBright** [comment]
buildkite/phobos has been running for 6 and a half hours now.

**Comment by WalterBright** [comment]
std.logger fails again.

**Comment by adamdruppe** [comment]
fyi this change broke the build of an important D user's commercial codebase. should prolly at least get a changelog entry.

**Comment by CyberShadow** [comment]
@WalterBright changed the unit test which was there to help prevent exactly that kind of breakage...

**Comment by adamdruppe** [comment]
to be fair it was an easy fix, took like 5 minutes to identify, fix, and trace back to this PR, and it isn't formally released yet anyway. but still would be nice if there were more justifications for these things written up in the changelog entries (which it doesn't look like there is one for this at all right now)

**Comment by CyberShadow** [comment]
For future reference (i.e. the inevitable proposed change which will involve similar breakages again), here is a link to a user complaint (to help evaluate said proposed changes):
https://forum.dlang.org/post/huhejrmzoelfqmdgxlxw@forum.dlang.org

**Pull request merged**